daemon/checkpoint.go

```diff
@@ -38,6 +39,25 @@ func (daemon *Daemon) CheckpointCreate(name string, config types.CheckpointCreat
 		return fmt.Errorf("Invalid checkpoint ID (%s), only %s are allowed", config.CheckpointID, validCheckpointNameChars)
 	}
 
+	var checkpointDirs string
+	containerList := []string{}
+
+	dir, _ := ioutil.ReadDir(containerDir)
+	for _, d := range dir {
+		containerList = append(containerList, d.Name())
+	}
+
+	for _, container := range containerList {
```

Comment:
why do you need this heavy iteration?
since the error is ` mkdir /var/lib/docker/containers/c4ba0c844c93a31db50dcda57fc0126fd17ec1c4a73cd550a501aac117685a6c/checkpoints/checkpoint2:
file exists`, you just consider the existence of `filepath.Join(daemon.repository, container.ID, "checkpoints", config.checkpointID)`